### ISSUE TITLE
App crashes randomly

### ISSUE THREAD

**Issue by Wojtaz0w** [NeoApplications/Neo-Store#401]
https://paste.crdroid.net/f8RnL6
https://paste.crdroid.net/xqPXW5
App crashed twice while it was running in background

https://paste.crdroid.net/WkpIqB
App crashed after clicking on notfication about new updates available

**Comment by machiav3lli** [comment]
Next you report please mention the version. It's version 0.9.12, right? If so, update to 0.9.13 then report again

**Comment by Wojtaz0w** [comment]
Sorry for not mentioning this! Notification crash was 0.9.12 but background was 0.9.13 I think

**Comment by HoyanMok** [comment]
I have got the same error in 0.9.13

```FATAL EXCEPTION: DefaultDispatcher-worker-5
Process: com.machiav3lli.fdroid, PID: 27760
java.lang.IllegalStateException: Mutex is not locked
	at kotlinx.coroutines.sync.MutexImpl.unlock(Mutex.kt:38)
	at com.machiav3lli.fdroid.service.SyncService$onDestroy$1.invokeSuspend(SyncService.kt:55)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:107)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:79)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@d4f0ef1, Dispatchers.Default]```

This happens randomly in background.

**Comment by HotarunIchijou** [comment]
Can accept this. I've got the same error that all you have.
https://bin.kv2.dev/~63ccc084e599576ea02b8696

This happens on 0.9.13 in background, on 0.9.12 didn't notice anything like that.

**Comment by WTechNinja** [comment]
I have the same thing, also on 0.9.13.
<details><summary>Stack Trace</summary><p>

```
FATAL EXCEPTION: DefaultDispatcher-worker-5
Process: com.machiav3lli.fdroid, PID: 315
java.lang.IllegalStateException: Mutex is not locked
	at kotlinx.coroutines.sync.MutexImpl.unlock(Mutex.kt:38)
	at com.machiav3lli.fdroid.service.SyncService$onDestroy$1.invokeSuspend(SyncService.kt:55)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:107)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:79)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@a0f78fe, Dispatchers.Default]
```

</p></details>

**Comment by pointydev** [comment]
Had multiple background crashes on 0.9.13 on `13 (TQ1A.221205.011)`.

**Comment by Quantenzitrone** [comment]
> Do you have auto-sync of repos set to the default 1 hour?

yep
and it seems to be connected to firefox klar having a security vulnerability while being disabled
at least it pops up hourly at tle same time

**Comment by machiav3lli** [comment]
For now, you may increase the auto-sync interval to 1440 minutes (a day) to avoid much crash dialogs. Am already testing possible solutions… and for now it seems to work

**Comment by machiav3lli** [comment]
Fixed on 0.9.14-alpha01 you can go ahead; test & report.

**Comment by pointydev** [comment]
> Fixed on 0.9.14-alpha01 you can go ahead; test & report.

Perhaps it would be a good idea to provide binaries for alphas so everyone is testing the same thing? Currently 0.9.14-alpha01 has no downloads beside source files.

**Comment by machiav3lli** [comment]
It's provided in the community groups and channel on Telegram & Matrix. These aren't and won't be provided here to avoid redundant reporting (GH-issues require investing more energy than simply being answered in the group as also other community members can be involved)

**Comment by Wojtaz0w** [comment]
I haven't seen any crashes since 0.9.14 alpha update. Auto sync works! :) 
![Screenshot_20230129-211915_crDroid Home.png](https://user-images.githubusercontent.com/104674625/215353701-531fa1c8-85b6-4443-ae0a-bdb15267938e.png)

**Comment by Wojtaz0w** [comment]
Issue fixed via 0.9.14-alpha1 
🙂
